katdal/test/test_datasources.py

```diff
@@ -0,0 +1,113 @@
+################################################################################
+# Copyright (c) 2018, National Research Foundation (Square Kilometre Array)
+#
+# Licensed under the BSD 3-Clause License (the "License"); you may not use
+# this file except in compliance with the License. You may obtain a copy
+# of the License at
+#
+#   https://opensource.org/licenses/BSD-3-Clause
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+"""Tests for :py:mod:`katdal.datasources`."""
+
+import tempfile
+import shutil
+import os
+
+import numpy as np
+from numpy.testing import assert_array_equal
+from nose.tools import assert_equal
+import dask.array as da
+
+from katdal.chunkstore import generate_chunks
+from katdal.chunkstore_npy import NpyFileChunkStore
+from katdal.datasources import ChunkStoreVisFlagsWeights
+
+
+def ramp(shape, offset=0.0, slope=1.0, dtype=np.float_):
+    """Generate a multidimensional ramp of values of the given dtype."""
+    x = offset + slope * np.arange(np.prod(shape), dtype=np.float_)
+    return x.astype(dtype).reshape(shape)
+
+
+def to_dask_array(x):
+    """Turn ndarray `x` into dask array with the standard vis-like chunking."""
+    n_corrprods = x.shape[2] if x.ndim >= 3 else x.shape[1] / 8
+    chunk_size = 4 * n_corrprods * 8
+    chunks = generate_chunks(x.shape, x.dtype, chunk_size,
+                             dims_to_split=(0, 1), power_of_two=True)
+    return da.from_array(x, chunks)
+
+
+def put_fake_dataset(store, base_name, shape):
+    """Write a fake dataset into the chunk store."""
+    data = {'correlator_data': ramp(shape, dtype=np.float32) * (1 - 1j),
+            'flags': np.zeros(shape, dtype=np.uint8),
```

Comment:
Maybe put some more interesting values in here to check that they carry through to the output.

Reply:
Changed zeros() to ones(). At least it caught a bug :-)